### PR TITLE
Document BIR spec and add BIR dump file flag

### PR DESCRIPTION
**Pull request by manuranga**

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 1]
```diff
@@ -149,6 +150,9 @@ public BuildCommand(Path userDir, PrintStream outStream, PrintStream errStream,
     @CommandLine.Option(names = "--dump-bir", hidden = true)
     private boolean dumpBIR;
 
+    @CommandLine.Option(names = "--dump-bir-file", hidden = true)
```
Comment (sameerajayasoma) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java:153]: How about --dump-bir-binary?

File: settings.gradle [hunk 2]
```diff
@@ -130,6 +130,7 @@ include(':build-config:checkstyle')
 include(':debug-adapter:debug-adapter-core')
 include(':debug-adapter:debug-adapter-cli')
 include(':ballerina-libs')
+include(':bir-spec')
```
Comment (sameerajayasoma) [inline settings.gradle:133]: Do we need to build this every time?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (Kishanthan) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java:153]: @sameerajayasoma , looks like with the option `--dump-bir-file` you have to give output file name/location. it is not a boolean flag apparently

**Comment by manuranga** [comment]
will be addressed in nballerina branch and merged to master later.

**Pull request closed without merging**